pkg/transport/utils/utils.go

```diff
@@ -0,0 +1,48 @@
+package utils
+
+import (
+	"bytes"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+)
+
+// WriteVarIntNum writes a variable-length integer to a buffer
+func WriteVarIntNum(writer *bytes.Buffer, num int) {
+	if num < 0 {
+		writer.WriteByte(0xff) // Representing -1
```

Comment:
Could you please take a look if GO-SDK is not implementing it in a better way?
I mean we need to add this library anyway and maybe we can use their implementation of VarInt